### DISCUSSION TITLE
Conditional parsing?

### DISCUSSION THREAD

**Discussion by Kein** [WerWolv/ImHex#830] [Pattern Language Q&A]
Consider the following
```
u8 Encrypted @ 0x00;
u32 count @ $;
u64 seeds[count] @ $;

if (!Encrypted)
{
    u128 Chunks[32] @ $;
}

MyStruct DescriptionData;
```

I only want to parse (and, consequently, to display) the data if specific condition is met and I'm failing to understand why the global variable, that is supposed to track the seeker position, cannot be used/operate inside scopes?

**Comment by WerWolv** [comment]
Hey
`if` statements in global scope currently work as if they're being used inside of a function. That's why you can't place a pattern in there. Instead of using the manual `@ $` syntax, I recommend you to instead make a struct that contains all that data, `Header` for example and add your patterns in there. That way you only need to place that struct once and the offsets of all other patterns just follow.

```rust
struct Header {
    u8 encrypted;
    u32 count;
    u64 seeds[count];

    if (!encrypted)
        u128 chunks[32];
};

Header header @ 0x00;
```

**Reply by Kein** [reply]
The issue is that now I have an extra `Header` entry in my pattern display that I need to expand and that clogs the view. The pattern window view already is not particularly good to begin with, and lacks customization, so wasting extra space is kind of a big deal here.

**Reply by WerWolv** [reply]
You can solve that by using the `[[inline]]` attribute on the Header struct. For the runtime that doesn't change anything but in the pattern data view, it will dump all of its content into the parent without further indentation

**Reply by Kein** [reply]
Ah, I see. is there a combination of attributes that can help get rid of extra-expand element for the array?
![image](https://user-images.githubusercontent.com/623756/203536887-3f7d9c8f-c51a-4e46-828e-5b2e7e11c109.png)

**Reply by WerWolv** [reply]
Not currently, no. I can maybe get rid of the `[0 ... 2]` tree element if there's only enough elements for one of them

**Reply by Kein** [reply]
Can it be done the way 010 does it? It really saves on space:
![image](https://user-images.githubusercontent.com/623756/203551433-136a0b22-d286-4c8d-8727-6e20fed93e1d.png)
